spec/models/spree/product_spec.rb

```diff
@@ -23,13 +23,11 @@ module Spree
       end
 
       it "does not save when master is invalid" do
-        s = create(:supplier_enterprise)
-        t = create(:taxon)
-        product = Product.new supplier_id: s.id, name: "Apples", price: 1, primary_taxon_id: t.id, variant_unit: "weight", variant_unit_scale: 1000, unit_value: 1
-        product.on_hand = "10,000"
-        expect(product.save).to be false
+        product = build(:product)
+        product.variant_unit = 'weight'
+        product.master.unit_value = nil
 
-        expect(product.errors[:count_on_hand]).to include "is not a number"
+        expect(product.save).to eq(false)
```

Comment:
it turns out `valid?` doesn't work for product and variant :sob: 

```ruby
[1] pry(#<RSpec::ExampleGroups::SpreeProduct::ValidationsAndDefaults>)> product.valid?
=> true
[2] pry(#<RSpec::ExampleGroups::SpreeProduct::ValidationsAndDefaults>)> product.master.valid?
=> true
[3] pry(#<RSpec::ExampleGroups::SpreeProduct::ValidationsAndDefaults>)> product.save!
ActiveRecord::RecordInvalid: Validation failed: Unit value can't be blank, Unit description can't be blank
from /home/pau/.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/activerecord-3.2.22.5/lib/active_record/validations.rb:56:in `save!'
[4] pry(#<RSpec::ExampleGroups::SpreeProduct::ValidationsAndDefaults>)>
```

God knows what we've broken...

Reply:
That is weird. Same in Spree 1? I guess we will notice if this is a problem... :see_no_evil: